### PR TITLE
Improve I18n pluralization to avoid errors with some calculators

### PR DESCRIPTION
**Pull request by dangerdogz**
This is a fix for issue #1164 

1) First commit is to add a preference containing the "magic number"
2) Second commit is to add the helper to wrap the usage of ActiveRecord::Naming
3) Third commit is find/replace to use the helpers

It is likely that we'll need to add more keys to fully support every pluralization calculators (slavic languages use more keys like : [ few, many ] and we only use one/other) but this address the underlying issue of trying to compare if a Symbol is less or greater than an integer which was causing crashes with the OneUptoTwoOther calculator used by the french language.

I opted for the preference/helper approach so the number can be updated without patching or monkey patching core, and the usage can be updated without editing every file from the backend but I can revert to a simpler usage if needed

EDIT : I also confirm that it fixes the issues raised by https://github.com/luukveenis/solidus_i18n/pull/6

Suggestions welcome

### CODE REVIEW

File: core/app/models/spree/app_configuration.rb [hunk 1]
```diff
@@ -274,6 +274,14 @@ class AppConfiguration < Preferences::Configuration
     #   @return [Boolean] Indicates if stock management can be restricted by location
     preference :can_restrict_stock_management, :boolean, default: false
 
+    # @!attribute [rw] i18n_generic_plural
+    #   This value is used as a count for the pluralization helpers related to I18n
+    #   ex: Spree::Order.model_name.human(count: Spree::I18N_GENERIC_PLURAL
+    #   Related to Solidus issue #1164, this is needed to avoid problems with
+    #   different pluralization calculators
+    #   @return [Integer] default: 2.1
+    preference :i18n_generic_plural, :integer, default: 2.1
```
Comment (jhawthorn) [inline core/app/models/spree/app_configuration.rb]: Is there a reason for this to be a preference? I think having it at `Spree::I18N_GENERIC_PLURAL` would probably be sufficient for everyone.

Comment (dangerdogz) [inline core/app/models/spree/app_configuration.rb]: Where should it live?

Comment (jhawthorn) [inline core/app/models/spree/app_configuration.rb]: Could be in here I suppose https://github.com/solidusio/solidus/blob/master/core/lib/spree/i18n.rb

Comment (dangerdogz) [inline core/app/models/spree/app_configuration.rb]: Noted, updating it

**Pull request closed without merging**